### PR TITLE
Re-write multi-model test in bash

### PR DESCRIPTION
**Pull request by SimonRichardson**
The multi-model test in python would fail because it would lose track
of which model to query. Considering we're attempting to remove these
python tests in favour of the bash ones, I took the time to do that.

The test is actually really easy and just creates a series of models and
ensures you can switch and talk to each one independently.

## QA steps

```sh
(cd tests && ./main.sh model)
```

### CODE REVIEW

File: tests/includes/juju.sh [hunk 1]
```diff
@@ -83,7 +83,7 @@ bootstrap() {
     output=${1}
     shift
 
-    rnd=$(head /dev/urandom | tr -dc a-z0-9 | head -c 8; echo '')
+    rnd=$(rnd_str)
```
Comment (achilleasa) [inline tests/includes/juju.sh:86]: How about `$RANDOM` since we use bash?

Comment (SimonRichardson) [reply tests/includes/juju.sh:86]: Because I want letters... ;-)

Comment (achilleasa) [reply tests/includes/juju.sh:86]: Why not EMOJIs! 🎆 🍕 🍺 ?

**Review by achilleasa — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**